src/HealthGPS.Console/configuration_parsing.cpp

```diff
@@ -0,0 +1,267 @@
+#include "configuration_parsing.h"
+#include "configuration_parsing_helpers.h"
+#include "jsonparser.h"
+
+#include <fmt/color.h>
+#include <stdexcept>
+#include <unordered_map>
+
+namespace host {
+using json = nlohmann::json;
+
+nlohmann::json get(const json &j, const std::string &key) {
+    try {
+        return j.at(key);
+    } catch (const std::exception &) {
+        fmt::print(fg(fmt::color::red), "Missing key \"{}\"\n", key);
+        throw ConfigurationError{fmt::format("Missing key \"{}\"", key)};
+    }
+}
+
+void rebase_valid_path(std::filesystem::path &path, const std::filesystem::path &base_dir) try {
+    if (path.is_relative()) {
+        path = std::filesystem::absolute(base_dir / path);
+    }
+
+    if (!std::filesystem::exists(path)) {
+        throw ConfigurationError{fmt::format("Path does not exist: {}", path.string())};
+    }
+} catch (const std::filesystem::filesystem_error &) {
+    throw ConfigurationError{fmt::format("OS error while reading path {}", path.string())};
+}
+
+// NOLINTNEXTLINE(bugprone-exception-escape)
+bool rebase_valid_path_to(const json &j, const std::string &key, std::filesystem::path &out,
+                          const std::filesystem::path &base_dir) noexcept {
+    if (!get_to(j, key, out)) {
+        return false;
+    }
+
+    try {
+        rebase_valid_path(out, base_dir);
+    } catch (const ConfigurationError &) {
+        fmt::print(fg(fmt::color::red), "Could not find file {}\n", out.string());
+        return false;
+    }
+
+    return true;
+}
+
+// NOLINTNEXTLINE(bugprone-exception-escape)
+void rebase_valid_path_to(const json &j, const std::string &key, std::filesystem::path &out,
```

Comment:
It's not, really. It's just that `out` might not be updated. Maybe in that case,
returning an empty `std::filesystem::path` wouldn't be terrible though.

Alternatively, this function could just return a `bool` indicating whether it's
succeeded, which is what I've done elsewhere (e.g. for the `get_to` function). That
would at least be consistent.

What do you think?

Reply:
Sorry, I meant that there's two `rebase_valid_path_to`, one on ln 34 which returns `success` as the return value, and another which returns `success` via a `bool` reference on ln 51. Is the latter necessary?

Yeah, for the sake of consistency, the boolean flag method is fine.

Reply:
The overload returning a `bool` isn't actually used anywhere, but I figured we might as
well keep it because we might do in future and it kind of makes sense to implement the
other overload using it anyway. That said, it's up to you. I could get rid of it and it
would save us a few lines of code.